timeline-chart/src/layer/time-graph-chart.ts

```diff
@@ -618,8 +625,26 @@ export class TimeGraphChart extends TimeGraphChartLayer {
             if (el instanceof TimeGraphStateComponent && !this.mousePanning && !this.mouseZooming) {
                 this.selectState(el.model);
             }
-            if (this.mouseInteractions && this.mouseInteractions.click) {
-                this.mouseInteractions.click(el, e);
+            
+            var clickCount = 1;
+            var self = this;
+            this._mouseClicks++;
+
+            if (this._mouseClicks === 1) {
+                this._recentlyClickedGlobal = cloneDeep(e.data.global);
+                this._dblClickTimer = setTimeout(function(){
+                    self._mouseClicks = 0;
+                    self._recentlyClickedGlobal = null;
+                }, this._dbClickTime);
+            } else if (this._recentlyClickedGlobal && (this._recentlyClickedGlobal.equals(e.data.global))){
```

Comment:
I would suggest that we only clear the timer and reset the click count if the global position is NOT the same one as click 1. In that case we should store the new position and restart another timer. Maybe this can be checked before the previous if-case so that you can reuse the code for first click. (click on a different point is then seen as a new first click, even if done quickly).

Then the 2nd, 3rd, etc. clicks on the same global position will just fall-through and send the event with the correct click count. No special case for count==2, it should just continue 2, 3, 4, ...